lib/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -506,15 +506,16 @@ sub delete {
         for (entities_with('ratings', take => 'model'));
 
     # Cancel any open edits the editor still has
-    my @edits = values %{ $self->c->model('Edit')->get_by_ids(
-        @{ $self->sql->select_single_column_array(
-            'SELECT id FROM edit WHERE editor = ? AND status = ?',
-            $editor_id, $STATUS_OPEN)
-       }
-    ) };
-
-    for my $edit (@edits) {
-        $self->c->model('Edit')->cancel($edit);
+    # We want to cancel the latest edits first, to make sure
+    # no conflicts happen that make some cancelling fail and all
+    # entities that should be autoremoved do get removed
+    my $ids = $self->sql->select_single_column_array(
+            'SELECT id FROM edit WHERE editor = ? AND status = ? ORDER BY open_time DESC',
```

Comment:
It should work since `open_time` [timestamp](https://www.postgresql.org/docs/9.6/datatype-datetime.html) is written down to the microsecond, but ordering by descending `id` would probably have worked too.